docs/cedarling/cedarling-policy-store.md

```diff
@@ -124,125 +88,184 @@ Here is a non-normative example of the `policies` field:
+    "openid_configuration_endpoint": "https://<trusted-issuer-hostname>/.well-known/openid-configuration",
+    "access_tokens": { 
+      "trusted": true,
+      "principlal_identifier": "jti",
+      ...
+    },
+    "id_tokens": { ... },
+    "userinfo_tokens": { ... },
+    "tx_tokens": { ... },
+  },
+  ...
+}
 ```
 
 - **name** : (*String*) The name of the trusted issuer.
 - **description** : (*String*) A brief description of the trusted issuer, providing context for administrators.
 - **openid_configuration_endpoint** : (*String*) The HTTPS URL for the OpenID Connect configuration endpoint (usually found at `/.well-known/openid-configuration`).
 - **identity_source** : (*Object*, *optional*) Metadata related to the tokens issued by this issuer.
+- **`access_tokens`, `id_tokens`, `userinfo_tokens`, `tx_tokens`** : (*Object*, *optional*) Metadata related to the tokens issued by this issuer. See schema [below](#token-entity-metadata-schema). **NOTE**: The `access_tokens` field contains the additional fields: `trusted` and `principal_identifier` in addition to the fields from it's `TokenEntityMetadata`.
 
-### Token Metadata Schema
+
+### Token Entity Metadata Schema
+
+The Token Entity Metadata Schema defines how tokens are mapped, parsed, and transformed within Cedarling. It allows you to specify how to extract user IDs, roles, and other claims from a token using customizable parsers.
 
 ```json
 {
-  "trusted": true|false
-  "principal_identifier": "some_user123",
-  "role_mapping": "role",
+  "token_type": {
+    "user_id": "<field name in token (e.g., 'email', 'sub', 'uid', etc.) or '' if not used>",
+    "role_mapping": "<field for role assignment (e.g., 'role', 'memberOf', etc.) or '' if not used>",
+    "claim_mapping": {
+      "mapping_target": {
+        "parser": "<type of parser ('regex' or 'json')>",
+        "type": "<type identifier (e.g., 'Acme::Email')>",
+        "...": "Additional configurations specific to the parser"
+      },
+    },
+  }
 }
 ```
 
-- **trusted** : (Boolean) The type of token
-- **principal_id** : (String) The claim used to create the Cedar entity associated with this token.
-- **role_mapping** : (String, *optional*) The claim used to create a role for the token. The default value of `role_mapping` is `role`. The claim can be string or array of string.
+**Fields**
+
+- **token_type:** The type of token being processed, such as `access_tokens`, `id_tokens`, `userinfo_tokens`, and `tx_tokens`.
+- **user_id (_Optional_):** The field in the token used to identify the user. If not needed, set to an empty string (`""`).
+- **role_mapping (_Optional_):** Indicates which field in the token should be used for role-based access control. If not needed, set to an empty string (`""`).
+- **claim_mapping:** Defines how to extract and transform specific claims from the token. Each claim can have its own parser (`regex` or `json`) and type (`Acme::Email`, `Acme::URI`, etc.).
```

Comment:
I believe it would be easier to understand if the regex parser in claim_mapping is explained with an example—showing how Cedarling would use the regex parser to populate the claim value from the token to the entity. Otherwise, the point remains unclear. Additionally, the json parser could also be explained with an example.

Reply:
I was only following this: https://github.com/JanssenProject/jans/wiki/Cedarling-Nativity-Plan. I honestly don't know how to use this as well so i couldn't provide an example on the behavior that Cedarling will do after it parsed the config.